systemd/systemd.go

```diff
@@ -216,66 +216,76 @@ func (*systemd) LogReader(serviceNames []string, n int, follow bool) (io.ReadClo
 
 var statusregex = regexp.MustCompile(`(?m)^(?:(.+?)=(.*)|(.*))?$`)
 
-type ServiceStatus struct {
-	Daemon          string
-	ServiceFileName string
-	Enabled         bool
-	Active          bool
+type UnitStatus struct {
+	Daemon   string
+	UnitName string
+	Enabled  bool
+	Active   bool
 }
 
-func (s *systemd) Status(serviceNames ...string) ([]*ServiceStatus, error) {
-	expected := []string{"Id", "Type", "ActiveState", "UnitFileState"}
-	cmd := make([]string, len(serviceNames)+2)
+func (s *systemd) Status(unitNames ...string) ([]*UnitStatus, error) {
+	// expected properties for timers and sockets
+	expectedBase := []string{"Id", "ActiveState", "UnitFileState"}
+	// for services and mounts
+	expectedAll := append(expectedBase, "Type")
+
+	cmd := make([]string, len(unitNames)+2)
 	cmd[0] = "show"
-	cmd[1] = "--property=" + strings.Join(expected, ",")
-	copy(cmd[2:], serviceNames)
+	// ask for all properties, regardless of unit type
+	cmd[1] = "--property=" + strings.Join(expectedAll, ",")
+	copy(cmd[2:], unitNames)
 	bs, err := systemctlCmd(cmd...)
 	if err != nil {
 		return nil, err
 	}
 
-	sts := make([]*ServiceStatus, 0, len(serviceNames))
-	cur := &ServiceStatus{}
+	sts := make([]*UnitStatus, 0, len(unitNames))
+	cur := &UnitStatus{}
 	seen := map[string]bool{}
 
 	for _, bs := range statusregex.FindAllSubmatch(bs, -1) {
 		if len(bs[0]) == 0 {
 			// systemctl separates data pertaining to particular services by an empty line
+			unitType := filepath.Ext(cur.UnitName)
+			expected := expectedAll
```

Comment:
The handling of expected attributes is a little bit confusing. Could we just have expectedSocket, expectedTimer, expectedMount, expectedService? I think it is fine to repeat something explicitly and pick the right _expected..._ variable based on the extension.